tool/bindings_generator/dart_main.dart

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:js_interop';
+import 'child_process_api.dart';
+import 'filesystem_api.dart';
+import 'generate_bindings.dart';
+import 'util.dart';
+
+// Generates DOM bindings for Dart.
+// TODO(joshualitt): This prototype has exposed a limitation of JS types on JS
+// backends.
+//   * Static interop doesn't seem to work on some types, atleast JSArrays. We
+//     should fix this.
+//  * We need to have a more specific type for JSPromise.
+// TODO(joshualitt): Find a way to generate bindings for JS builtins. This will
+// probably involve parsing the TC39 spec.
+
+void runDartFormat(String dir) {
```

Comment:
FYI @srujzs `DartFormat` gave me some issues when I ran it on Node, something with `Uri.base` or something. This may be fixable, but its easy to just call out to the host too.

Reply:
Weird, but okay. I do want us to move to source at some point, so mind adding a TODO? It'll be useful to mention what issues you came across for the future.

Reply:
Whoops, good thing I looked a bit into this. We needed `self.location` set in our preamble.